manager/controllers/app/data_catalog_interface.go

```diff
@@ -21,7 +21,7 @@ import (
 // - an error if happened
 // - the new asset identifier
 func (r *FybrikApplicationReconciler) RegisterAsset(assetID string, catalogID string,
-	info *fapp.DatasetDetails, input *fapp.FybrikApplication) (string, error) {
+	info *fapp.DatasetDetails, input *fapp.FybrikApplication) (*datacatalog.CreateAssetResponse, error) {
```

Comment:
Why are you changing the signature? The original behavior was to register the asset and return the registered asset id or an error. Why not do the validation here and return the same after validation.

Reply:
I'll do that Thanks, I mistakenly thought the functions do not belong to the same class.